qiskit/pulse/parameter_manager.py

```diff
@@ -424,7 +425,24 @@ def _unroll_param_dict(
                     )
                 out.update(zip(parameter, value))
             elif isinstance(parameter, str):
-                out[self.get_parameters(parameter)] = value
+                for param in self.parameters:
+                    if param.name == parameter:
+                        out[param] = value
+                    elif (
+                        isinstance(param, ParameterVectorElement) and param.vector.name == parameter
```

Comment:
I find this part to be getting a little complicated. Could we lift the string handling to before this loop? So in a preceding loop, loop over `parameter_binds` and make a new copy that has all `Parameter` and `ParameterVector` keys by replacing any strings with the appropriately named matches?